Source/Core/Core/NetPlayServer.cpp

```diff
@@ -1244,6 +1299,84 @@ bool NetPlayServer::SyncSaveData()
   return true;
 }
 
+bool NetPlayServer::SyncGeckoCodes()
+{
+  // Initialize Number of Synced Players
+  m_gecko_codes_synced_players = 0;
+
+  // Get Game Path
+  const auto game = m_dialog->FindGameFile(m_selected_game);
+  if (game == nullptr)
+  {
+    PanicAlertT("Selected game doesn't exist in game list!");
+    return false;
+  }
+
+  // Find all INI files
+  std::string game_id = game->GetGameID();
+  std::optional<u16> revision = game->GetRevision();
+  IniFile globalIni;
+  for (const std::string& filename : ConfigLoaders::GetGameIniFilenames(game_id, revision))
+    globalIni.Load(File::GetSysDirectory() + GAMESETTINGS_DIR DIR_SEP + filename, true);
+  IniFile localIni;
+  for (const std::string& filename : ConfigLoaders::GetGameIniFilenames(game_id, revision))
+    localIni.Load(File::GetUserPath(D_GAMESETTINGS_IDX) + filename, true);
+
+  // Create a Gecko Code Vector with just the active codes
+  std::vector<Gecko::GeckoCode> s_active_codes =
+      Gecko::SetAndReturnActiveCodes(Gecko::LoadCodes(globalIni, localIni));
+
+  // Determine Codelist Size
+  u16 codelines = 0;
+  for (const Gecko::GeckoCode& active_code : s_active_codes)
+  {
+    NOTICE_LOG(ACTIONREPLAY, "Indexing %s", active_code.name.c_str());
+    for (const Gecko::GeckoCode::Code& code : active_code.codes)
+    {
+      NOTICE_LOG(ACTIONREPLAY, "%08x %08x", code.address, code.data);
+      codelines++;
+    }
+  }
+
+  // Output codelines to send
+  NOTICE_LOG(ACTIONREPLAY, "Sending %d codelines", codelines);
+
+  // Wait to start the game. This is done here instead of when the function returns because
+  // in most cases, the clients will respond as finished before this function returns, causing
+  // it to hang since the pending start flag will be set to true AFTER the response set it to false.
+  m_start_pending = true;
```

Comment:
So just move setting this variable above the function call instead of below it, and reset it to false in the failure case.  This is an odd place to put this.

You can move it above the call for save sync as well, to avoid the possibility of this race condition (though I've never encountered it there).